Description: Tighten long header packet length check for 32-bit platform
Author: Tatsuhiro Tsujikawa <tatsuhiro.t@gmail.com>
Origin: upstream, https://github.com/ngtcp2/ngtcp2/commit/2e47b5dbac26ee280f832046186bdcd0883dc096
Bug: https://github.com/ngtcp2/ngtcp2/pull/2280
Last-Update: 2026-10-05
---
diff --git a/lib/ngtcp2_pkt.c b/lib/ngtcp2_pkt.c
index d1461458..f430f43d 100644
--- a/lib/ngtcp2_pkt.c
+++ b/lib/ngtcp2_pkt.c
@@ -307,7 +307,7 @@ ngtcp2_ssize ngtcp2_pkt_decode_hd_long(ngtcp2_pkt_hd *dest, const uint8_t *pkt,

     ngtcp2_get_uvarint(&vi, p);
 #if SIZE_MAX < UINT64_MAX
-    if (vi > SIZE_MAX) {
+    if (vi > SIZE_MAX / 2) {
       return NGTCP2_ERR_INVALID_ARGUMENT;
     }
 #endif /* SIZE_MAX < UINT64_MAX */
